.github/workflows/ci.yml

```
@@ -194,6 +201,111 @@ jobs:
           if-no-files-found: error
           retention-days: 14
 
+  Test-Suite-Fork:
+    needs: [Precheck]
+    environment: Testing
+    if: ((startsWith(github.ref, 'refs/tags/') ||
+      github.ref == 'refs/heads/main' ||
+      needs.precheck.outputs.srcfileschanged == 'true') &&
+      github.event.pull_request.head.repo.full_name == github.repository) != true # skip when trusted workflow runs
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: src
+    steps:
+      - name: Fail check
+        if: ${{ (github.event_name == 'repository_dispatch' &&
+          github.event.client_payload.slash_command.sha != '' &&
+          contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.sha)) != true }}
+        run: exit 1
```

@meain here we check if its NOT a repo_dispatch event - exit with an error

---

Can we just simplify the check to `github.event_name != 'repository_dispatch'`? Also, maybe we can add a message in the `run` section so as to give a hint to contributors as to what is happening. Something like below:
```suggestion
        run: |
        	echo "Workflow requires approval from a maintainer to run. It will be automatically rerun on approval."
        	exit 1
```